src/resolve/resolved-dns-trust-anchor.c

```diff
@@ -274,6 +275,14 @@ static int dns_trust_anchor_load_positive(DnsTrustAnchor *d, const char *path, u
                         return -EINVAL;
                 }
 
+                if (isempty(p)) {
+                        log_warning("Missing DS parametes on line %s:%u", path, line);
+                        return -EINVAL;
+                }
+
+                digest = strdupa(p);
+                digest = delete_chars(digest, WHITESPACE);
```

Comment:
Hmm, I wonder if it wouldn't be better to tweak `unhexmem` instead to ignore whitespace. Possibly as a flag, in case there are places where we want to preserve it. It just seems that in majority of cases we don't care about whitespace at all.

Reply:
`root_hash_load` in `src/shared/dissect-image.c` would imho benefit from this too: it seems unnecessarily strict to forbid trailing whitespace in `.roothash`.

Reply:
OK. I will update this.